pkg/api/types.go

```diff
@@ -180,6 +180,139 @@ type VolumeSource struct {
+}
+
+type PersistentVolumeSpec struct {
+	// Resources represents the actual resources of the volume
+	Resources ResourceList `json:"capacity,omitempty`
+	// Source represents the location and type of a volume to mount.
+	// AccessModeTypes are inferred from the Source.
+	Source PersistentVolumeSource `json:",inline"`
+}
+
+type PersistentVolumeStatus struct {
+	// Phase represents the current phase of the PersistentVolume
+	Phase StoragePhase `json:"phase,omitempty"`
+}
+
+type PersistentVolumeList struct {
+	TypeMeta `json:",inline"`
+	ListMeta `json:"metadata,omitempty"`
+	Items    []PersistentVolume `json:"items,omitempty"`
+}
+
+// a PersistentVolumeClaim is a user's request for and claim to a persistent volume
+type PersistentVolumeClaim struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the volume
+	Spec PersistentVolumeClaimSpec `json:"spec,omitempty"`
+
+	// Status represents the current information about a persistent volume.
+	Status PersistentVolumeClaimStatus `json:"status,omitempty"`
+}
+
+type PersistentVolumeClaimList struct {
+	TypeMeta `json:",inline"`
+	ListMeta `json:"metadata,omitempty"`
+	Items    []PersistentVolumeClaim `json:"items,omitempty"`
+}
+
+// a PersistentVolumeClaimSpec describes the common attributes of storage devices
+// and allows a Source for provider-specific attributes
+type PersistentVolumeClaimSpec struct {
+	// Contains the types of access modes desired
+	AccessModes []AccessModeType `json:"accessModes,omitempty"`
+	// Resources represents the minimum resources desired
+	Resources ResourceRequirements `json:"resources,omitempty"`
+}
+
+type PersistentVolumeClaimStatus struct {
+	// Phase represents the current phase of PersistentVolumeClaim
+	Phase StoragePhase `json:"phase,omitempty"`
+	// AccessModes contains all ways the volume backing the PVC can be mounted
+	AccessModes []AccessModeType `json:"accessModes,omitempty`
+	// Represents the actual resources of the underlying volume
+	Resources ResourceList `json:"requests,omitempty"`
+	// VolumeRef is a reference to the PersistentVolume bound to the PersistentVolumeClaim
+	VolumeRef *ObjectReference `json:"volumeRef,omitempty"`
+}
+
+type AccessModeType string
+
+const (
+	ReadWriteOnce AccessModeType = "ReadWriteOnce"
+	ReadOnlyMany  AccessModeType = "ReadOnlyMany"
+	ReadWriteMany AccessModeType = "ReadWriteMany"
+)
+
+type StoragePhase string
+
+const (
+	// used for PersistentVolumes and PersistentVolumeClaims that are not yet bound
+	StoragePending StoragePhase = "Pending"
+	// used for PersistentVolumes and PersistentVolumeClaims that are bound
+	StorageBound StoragePhase = "Bound"
+	// used for PersistentVolumes where the bound PersistentVolumeClaim was deleted
+	StorageRelease StoragePhase = "Released"
+)
+
+type PersistentVolumeClaimVolumeSource struct {
+	// AccessMode expresses how a volume should be mounted.
+	AccessMode AccessModeType `json:"accessMode,omitempty"`
```

Comment:
I think you're right.   The implied AccessMode would come from the bool and the usage (used in many pods or just one).  I made it explicit here, though, because everything else is explicit.